includes/constants/class-base-constant.php

```diff
@@ -17,7 +17,7 @@
 /**
  * Base constant class to hold common logic for all constants.
  */
-abstract class Base_Constant {
+abstract class Base_Constant implements \JsonSerializable {
```

Comment:
Nitpick: though not directly related to this issue, but I think it makes sense to add Stringable here for the existing `_toString()` method.

> As of PHP 8.0.0, any class that contains a __toString() method will also implicitly implement the Stringable interface, and will thus pass type checks for that interface. Explicitly implementing the interface anyway is recommended.
> -- Source https://www.php.net/manual/en/language.oop5.magic.php#object.tostring

```suggestion
abstract class Base_Constant implements \JsonSerializable, \Stringable {
```

Reply:
@james-allan - Oops...sorry that I did not check which PHP versions supporting this interface. It turns out that interface `Stringable` is only available since PHP8. Therefore, it's safe to not include it in your PR https://www.php.net/manual/en/class.stringable.php

Reply:
Oh good find. That would have been awkward. Reverted in [ac4daea](https://github.com/Automattic/woocommerce-payments/pull/6753/commits/ac4daea027fcf7af622d4c78cf15fd4b98cb4b97)